session-loader/Development/IDE/Session.hs

```diff
@@ -118,9 +118,12 @@ loadSession dir = do
         -- files in the project so that `knownFiles` can learn about them and
         -- we can generate a complete module graph
     let extendKnownTargets newTargets = do
-          knownTargets <- forM newTargets $ \TargetDetails{..} -> do
-            found <- filterM (IO.doesFileExist . fromNormalizedFilePath) targetLocations
-            return (targetTarget, found)
+          knownTargets <- forM newTargets $ \TargetDetails{..} ->
+            case targetTarget of
+              TargetFile f -> pure (targetTarget, [f])
```

Comment:
Don't want to call `doesFileExist` here because the file might be purely virtual.

Reply:
we can call `GetFileExists` which should work fine for virtual files, if you think this can be a problem in practice